tools/extra/pac/fpgabist/dma/fpga_dma_test.c

```diff
@@ -54,6 +61,132 @@ static int err_cnt;
       goto label;\
     } \
   } while (0)
+/*
+ * Global configuration of bus, set during parse_args()
+ * */
+struct config{
+  struct target {
+	  int bus;
+  } target;
+} config = {
+	.target = {
+		.bus = -1
+	}
+};
+
+void help(void)
+{
+	printf(
+"\n"
+"fpga_dma_test \n"
+"Usage:\n"
+"\n"
+"               0          Hardware\n"
+"               1          Simulation only\n"
+"              -B, --bus   Set target bus number (OPTIONAL)\n"
+"\n"
+                   );
+}
+
+
+/*
+ * Parse command line arguments
+ */
+#define GETOPT_STRING "B"
+int parse_args(int argc, char *argv[])
+{
+	struct option longopts[] = {
+		
+		{"bus",          required_argument, NULL, 'B'}
+	};
+	
+	int getopt_ret;
+	int option_index;
+	char *endptr = NULL;
+
+	while (-1 != (getopt_ret = getopt_long(argc, argv, GETOPT_STRING, 
+						longopts, &option_index))) {
+		const char *tmp_optarg = optarg;
+		
+		if ((optarg) && ('=' == *tmp_optarg)){
+			++tmp_optarg;
+		}
+		
+		switch (getopt_ret) {
+		case 'B':   /* bus */
+			if (NULL == tmp_optarg)
+				break;
+			endptr = NULL;
+			config.target.bus = (int) strtoul(tmp_optarg, &endptr, 0);
```

Comment:
You should use safestring versions of strtoul and strlen if they are available

Reply:
I couldn't find the safestring version of strtoul but I did find the safestring version of strlen which is strnlen. Do you have any other suggestions for strtoul?

Reply:
Ok. No problem. Use strtoul in that case.